Mathlib/SetTheory/Cardinal/Basic.lean

```diff
@@ -917,6 +924,10 @@ def prod {ι : Type u} (f : ι → Cardinal) : Cardinal :=
 theorem mk_pi {ι : Type u} (α : ι → Type v) : #(∀ i, α i) = prod fun i => #(α i) :=
   mk_congr <| Equiv.piCongrRight fun _ => outMkEquiv.symm
 
+theorem mk_pi_eq {ι : Type u} {f g : ι → Type v} (h : ∀ i, #(f i) = #(g i)) :
+    #(Π i, f i) = #(Π i, g i) := by
+  simp_all only [mk_pi]
```

Comment:
I'm inclined to say this shouldn't be a lemma since it does indeed follow easily from `mk_pi`. But maybe there's a bounded quantification one which doesn't follow quite so easily - `#(Π i ∈ s, f i) = #(Π i ∈ s, g i)` (with `h` adjusted to also only be in `s`)? I'm not sure, what do you think?

Reply:
I like having these kinds of lemmas because in an actual use case `simp_all only [mk_pi]` won't reduce the goal this way, since it won't be this clean. Why not have this so you can easily tell lean exactly what it should do?
I'm usually leaning towards adding lemmas like this just to prevent cumulative clutter.

The generalization looks like a good idea, I'll add it.

Reply:
That's fair. Let's give it the name `mk_pi_congr` then.